server/src/main/java/org/elasticsearch/index/seqno/ReplicationTracker.java

```diff
@@ -916,6 +919,47 @@ public ReplicationGroup getReplicationGroup() {
         return replicationGroup;
     }
 
+    /**
+     * Returns the pending replication actions for the shard.
+     *
+     * @return the pending replication actions
+     */
+    public PendingReplicationActions getPendingReplicationActions() {
+        assert primaryMode;
+        return pendingReplication;
+    }
+
+    private void updateReplicationGroupAndNotify(boolean replicasMightHaveChanged) {
```

Comment:
I think this optimization (replicasMightHaveChanged) is unnecessary (and easy to get wrong). The methods that call `updateReplicationGroupAndNotify` are rarely called, so let's omit this.